README.md

```diff
@@ -168,305 +116,247 @@ Typical Java threads
+There are various adapters to abstract how data is picked up by Emissary.  Ultimately the data is received in PickUpPlace and is broken out
+ into BaseDataObject sessions.  PickUpPlace then pulls a MobileAgent from the thread pool and hands off the session for processing in [assignToPooledAgent()](src/main/java/emissary/pickup/PickUpPlace.java#L719).
+
+### Processing Routing
+
+MobileAgent walks a given session through all the relevant places.
+
+There are two types of MobileAgent:
+* [MobileAgent](src/main/java/emissary/core/MobileAgent.java) - Used for processing a single payload
+* [HDMobileAgent](src/main/java/emissary/core/HDMobileAgent.java) - Used for processing a bundle of payloads
+
+Processing starts when PickUpPlace's assignToPooledAgent() calls [agent.go(payload, startingLocation)](src/main/java/emissary/pickup/PickUpPlace.java#L751)
 
-#### Processing Routing
+The MobileAgent processing loop is handled in [agentControl()](src/main/java/emissary/core/MobileAgent.java#L249)
 
-MobileAgent walks a given session through all the relevant places.  This happens by 
-MobileAgent#go() -> MobileAgent#agentControl() -> MobileAgent#getNextKey() -> MobileAgent#getNextKeyFromDirectory which 
-ultimately hooks into DirectoryPlace#nextKeys().
+To identify which ServiceProviderPlace the MobileAgent will visit next, it calls [getNextKey()](src/main/java/emissary/core/MobileAgent.java#L272)
 
-#### Workflow
+### Workflow and Stages
 
-The Emissary data driven workflow has stages. These are specified by emissary.core.Stage. The stages of the workflow 
+The Emissary data-driven workflow processes in stages. These are specified by [emissary.core.Stage](src/main/java/emissary/core/Stage.java). The stages of the workflow 
 are used to control certain aspects of unwrapping and processing and help to ensure that the workflow can always make 
 progress on the task at hand.
 
 There are guidelines on how the workflow stages should be used. These guidelines are not currently enforced but 
 deviations from these guidelines might be an indication that more thought should be applied.
 
-##### Study
-
-The name for this stage came from the idea behind [Perl's Regex study](http://perldoc.perl.org/functions/study.html) 
+Stage Name    | Description                  | Run in Parallel?
+:-------------|:-----------------------------|:----------------:
+STUDY         | Prepare, coordinate idents   | No
+ID            | Identification phase         | No
+COORDINATE    | Coordinate processing        | No
+PRETRANSFORM  | Before transform hook        | Yes
+TRANSFORM     | Transformation phase         | No
+POSTTRANSFORM | After transform hook         | Yes
+ANALYZE       | Analysis/metadata generation | Yes
+VERIFY        | Verify output                | No
+IO            | Output                       | No
+REVIEW        | Finish off                   | No
+
+* #### Study
```

Comment:
Bulleted phases to make it clearer that these were subsections of the Workflow section